montyDSL.qmd

```diff
@@ -0,0 +1,112 @@
+# The monty DSL {#sec-monty-dsl}
+
+```{r}
+#| include: false
+source("common.R")
+```
+
+The `monty` DSL provides a more intuitive way to define statistical models with `monty`. It is currently relatively basic and focus on providing support for defining priors in Bayesian models. It fully support differentiability allowing to use gradient based samplers on these models.
+
+```{r}
+library(monty)
+```
+
+## Some simple examples
+
+In chapter 4 of Statistical Rethinking, we build a regression model of height with parameters $\alpha$, $\beta$ and $\sigma$.  We can define the model for the prior probability of this model in monty by running
+
+```{r}
+prior <- monty_dsl({
+  alpha ~ Normal(178, 20)
+  beta ~ Normal(0, 10)
+  sigma ~ Uniform(0, 50)
+})
+```
+
+This will define a new `monty_model()` object that represents the prior, but with all the bits that we might need depending on how we want to use it:
+
+We have model parameters
+
+```{r}
+prior$parameters
+```
+
+These are defined in the order that they appear in your definition (so `alpha` is first and `sigma` is last)
+
+We can compute the domain for your model:
+
+```{r}
+prior$domain
+```
+
+We can draw samples from the model if we provide a `monty_rng` object
+
+```{r}
+rng <- monty_rng$new()
+theta <- prior$direct_sample(rng)
+theta
+```
+
+We can compute the (log) density at a point in parameter space
+
+```{r}
+prior$density(theta)
+```
+
+The computed properties for the model are:
+
+```{r}
+prior$properties
+```
+
+## Calculations in the DSL
```

Comment:
This whole section is verbatim from the dsl vignette - there's no point duplicating everything, we can just point to it.  So this should be building to something bigger within this book, and until it does perhaps just drop it.

Reply:
Might be worth just highlighting the example where `c` depends on `a` and `b` as an illustration of how it allows dependent prior distributions?

Reply:
Could even just have a simple two parameter model where `b` depends on `a`